nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -29,28 +34,54 @@ public static NsisExecutable createNsisExecutable(GenericFactory factory,
 	private void initNsisExecutable(GenericFactory factory, ByteProvider bp,
 			SectionLayout layout) throws IOException {
 		this.reader = new FactoryBundledWithBinaryReader(factory, bp, true);
-		this.header_offset = -1;
+		this.headerOffset = -1;
 		findHeaderOffset();
-		// TODO Init sections and headers
+		initScriptHeader(bp);
 	}
 
 	private void findHeaderOffset() throws IOException {
-		if (this.header_offset == -1) {
+		if (this.headerOffset == -1) {
```

Comment:
Consider `findHeaderOffset` to return the offset, not conditionally write it to `this.headerOffset`. It will make the function cleaner and the variable assignment clearer.

Reply:
Good idea, updated